### ISSUE TITLE
Argo S3 artifacts don't work when minio client needs to assume role

### ISSUE THREAD

**Issue by firecast** [argoproj/argo-workflows#11723]
### Pre-requisites

- [X] I have double-checked my configuration
- [X] I can confirm the issues exists when I tested with `:latest`
- [X] I'd like to contribute the fix myself (see [contributing guide](https://github.com/argoproj/argo-workflows/blob/master/docs/CONTRIBUTING.md))

### What happened/what you expected to happen?

Expected:
- S3 artifacts would be fetched where the IAM role had to assume another role

Happened:
- Pod wait container was not able to fetch artifacts and gave error
`http://169.254.169.254/latest/api/token": context deadline exceeded`

This happens if the EC2 instances is not enabled for IMDSv2 and was introduced as part of [this](https://github.com/minio/minio-go/pull/1682) PR on minio-go and was merged in Argo v3.4.0. It has since been [fixed](https://github.com/minio/minio-go/issues/1866) and released as a BugFix version [v7.0.63](https://github.com/minio/minio-go/releases/tag/v7.0.63). We will need to just update the the minio dependency to fix it.

### Version

v3.4.9 and latest

### Paste a small workflow that reproduces the issue. We must be able to run the workflow; don't enter a workflows that uses private images.

```YAML
apiVersion: argoproj.io/v1alpha1
kind: WorkflowTemplate
metadata:
  name: ghost-transform
  namespace: argo
spec:
  podGC:
    strategy: OnPodSuccess
    deleteDelayDuration: 500s
  templates:
    - name: tails
      script:
        image: python:alpine3.6
        command: [ python ]
        source: |
          with open("result.txt", "w") as f:
            f.write("it was tails")
      inputs:
        artifacts:
        - name: input
          path: "/tmp/inputs"
      outputs:
        artifacts:
          - name: result
            path: /result.txt

    - name: main
      dag:
        tasks:
          - name: run
            template: tails
            arguments:
              artifacts:
                - name: input
                  s3: # access to this artifact required the minio client to assume a role
                    bucket: "some-non-default-artifact-bucket"
                    key: "some-prefix"
                    region: "eu-central-1"
                    endpoint: "s3.eu-central-1.amazonaws.com"
                
  entrypoint: main
```

### Logs from the workflow controller

```text
Not relevant
```

### Logs from in your workflow's wait container

```text
http://169.254.169.254/latest/api/token": context deadline exceeded
```

**Comment by terrytangyuan** [comment]
Would you like to submit a PR to upgrade it?

**Comment by agilgur5** [comment]
This was updated in `argoproj/pkg` as part of https://github.com/argoproj/pkg/pull/492, so just `pkg` needs a bump. May happen as part of https://github.com/argoproj/argo-workflows/pull/11704#issuecomment-1704421259

**Comment by agilgur5** [comment]
Should be fixed by #11704 now
